### PR TITLE
ci: run clang-tidy on PR diff and post suggestions

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
For our betterment.

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [x] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [x] Changes have been communicated to collaborators @veprbl @c-dilks 

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No.

### Does this PR change default behavior?
Let's hope so. Wait, this probably doesn't refer to the behavior of developers...

### CODE REVIEW

File: .github/workflows/linux-eic-shell.yml [hunk 1]
```diff
@@ -104,13 +104,13 @@ jobs:
         name: build-clang-eic-shell-Debug
         path: build/
     - uses: eic/run-cvmfs-osg-eic-shell@main
-      if: ${{ github.event_name == 'pull_request'}}
+      if: ${{ github.event_name == 'pull_request' && ! contains(github.event.comment.body, 'run_clang_tidy') }}
```
Comment (wdconinc) [inline .github/workflows/linux-eic-shell.yml:107]: Nevermind, this requires `on: issue_comment` which we don't want.
```suggestion
      if: ${{ github.event_name == 'pull_request' }}
```

File: .github/workflows/linux-eic-shell.yml [hunk 2]
```diff
@@ -104,13 +104,13 @@ jobs:
...
       with:
         platform-release: "jug_xl:nightly"
         run: |
           git diff ${{github.event.pull_request.base.sha}} | clang-tidy-diff -p 1 -path build -quiet -export-fixes clang_tidy_fixes.yml -extra-arg='-std=c++17' -clang-tidy-binary run-clang-tidy
     - uses: eic/run-cvmfs-osg-eic-shell@main
-      if: ${{ github.event_name == 'push'}}
+      if: ${{ github.event_name == 'push' || ( github.event_name == 'pull_request' && contains(github.event.comment.body, 'run_clang_tidy') ) }}
```
Comment (wdconinc) [inline .github/workflows/linux-eic-shell.yml:113]:
```suggestion
      if: ${{ github.event_name == 'push' }}
```

File: .github/workflows/linux-eic-shell.yml [hunk 3]
```diff
@@ -104,13 +104,13 @@ jobs:
...
     - uses: eic/run-cvmfs-osg-eic-shell@main
-      if: ${{ github.event_name == 'pull_request'}}
+      if: ${{ github.event_name == 'pull_request' }}
       with:
         platform-release: "jug_xl:nightly"
         run: |
           git diff ${{github.event.pull_request.base.sha}} | clang-tidy-diff -p 1 -path build -quiet -export-fixes clang_tidy_fixes.yml -extra-arg='-std=c++17' -clang-tidy-binary run-clang-tidy
     - uses: eic/run-cvmfs-osg-eic-shell@main
-      if: ${{ github.event_name == 'push'}}
+      if: ${{ github.event_name == 'push' }}
```
Comment (wdconinc) [inline .github/workflows/linux-eic-shell.yml:113]:
```suggestion
      if: ${{ github.event_name == 'push' || ( github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'clang-tidy') ) }}
```

File: .github/workflows/linux-eic-shell.yml [hunk 4]
```diff
@@ -119,6 +119,13 @@ jobs:
       with:
         name: clang-tidy-fixes.yml
         path: clang_tidy_fixes.yml
+    - uses: platisd/clang-tidy-pr-comments@v1.4.0
```
Comment (wdconinc) [inline .github/workflows/linux-eic-shell.yml:122]:
```suggestion
    - uses: platisd/clang-tidy-pr-comments@1.4.0
```

**Comment by wdconinc** [comment]
The clang-tidy label won't actually do anything. GitHub Actions only allows code suggestions in the context of the modified code. So, even forcing a full clang-tidy run doesn't allow this action to post the suggestions in areas outside the initial diff contexts. We would be able to create a new PR against this branch with all changes, but then there are no suggestions, only all fixes applied.

**Comment by veprbl** [comment]
It works: #754 
We could drop the label, and see if it can be enabled unconditionally. Probably should check all clang-tidy reports for unwanted suggestions first.

**Comment by wdconinc** [comment]
I'm worried about
>Build against eic-shell / clang-tidy (pull_request) Successful in 79m

That really slows down the pipeline substantially.

Most of the issues are even in system headers, not in the headers we care about. I think there's a way to filter them from being analyzed, but I never managed to make that work.

File: .github/workflows/linux-eic-shell.yml [hunk 5]
```diff
@@ -104,13 +104,13 @@ jobs:
...
     - uses: eic/run-cvmfs-osg-eic-shell@main
-      if: ${{ github.event_name == 'pull_request'}}
+      if: ${{ github.event_name == 'pull_request' }}
       with:
         platform-release: "jug_xl:nightly"
         run: |
           git diff ${{github.event.pull_request.base.sha}} | clang-tidy-diff -p 1 -path build -quiet -export-fixes clang_tidy_fixes.yml -extra-arg='-std=c++17' -clang-tidy-binary run-clang-tidy
     - uses: eic/run-cvmfs-osg-eic-shell@main
-      if: ${{ github.event_name == 'push'}}
+      if: ${{ github.event_name == 'push' || ( github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'clang-tidy') ) }}
```
Comment (wdconinc) [inline .github/workflows/linux-eic-shell.yml:113]:
```suggestion
      if: ${{ github.event_name == 'push' }}
```

**Comment by wdconinc** [comment]
So, I think unconditional is fine, as long as it's just based on the git diff. In that case

> Build against eic-shell / clang-tidy (pull_request) Successful in 6m

The label was an attempt to get more of the code qa-tested... Foolhardy.

**Comment by veprbl** [comment]
In #754 it's

> Build against eic-shell / clang-tidy (pull_request) Successful in 6m

**Review by veprbl — DISMISSED** [review]

**Comment by veprbl** [comment]
To reiterate. My understanding, your plan was to see if we can do global checks in special PRs with the clang-tidy tags. Instead, if everything goes well, we'll be getting suggestions on each submitted PR, possibly asking to address pre-existing issues.

File: .github/workflows/linux-eic-shell.yml [hunk 6]
```diff
@@ -104,13 +104,13 @@ jobs:
         name: build-clang-eic-shell-Debug
         path: build/
     - uses: eic/run-cvmfs-osg-eic-shell@main
-      if: ${{ github.event_name == 'pull_request'}}
+      if: ${{ github.event_name == 'pull_request' }}
       with:
         platform-release: "jug_xl:nightly"
         run: |
           git diff ${{github.event.pull_request.base.sha}} | clang-tidy-diff -p 1 -path build -quiet -export-fixes clang_tidy_fixes.yml -extra-arg='-std=c++17' -clang-tidy-binary run-clang-tidy
```
Comment (veprbl) [inline .github/workflows/linux-eic-shell.yml:111]:
```suggestion
          git diff ${{github.event.pull_request.base.sha}} | clang-tidy-diff -p 1 -path build -quiet -export-fixes clang_tidy_fixes.yml -extra-arg='-std=c++17' -checks='-*,bugprone-*,-bugprone-narrowing-conversions' -clang-tidy-binary run-clang-tidy
```

**Review by veprbl — APPROVED** [review]

**Pull request merged**